oggm/utils/_funcs.py

```diff
@@ -753,8 +762,19 @@ def find_inversion_calving(gdir, water_depth=1, max_ite=30,
             # from a non-calving glacier)
             f_calving = 0
         elif i == 1:
-            # Second call we set a small positive calving to start with
-            out = calving_flux_from_depth(gdir, water_depth=water_depth)
+            if fix_water_depth:
+                # We fix the water and calculate a pre-calving
+                # assuming the first thicknes to be:
+                cl = gdir.read_pickle('inversion_output')[-1]
+                t_altitude = cl['hgt'][-1]
+                thick = t_altitude + 1
+                out = calving_flux_from_depth(gdir,
+                                              water_depth=water_depth,
+                                              thick=thick,
+                                              fix_water_depth=True)
```

Comment:
I don't understand this part. If we fix the water depth, shouldn't the original thickness be `water_depth + 1`? Or do we allow negative free-board? In any case, I don't think that `t_altitude` should play any role here, because this is what becomes a free parameter in the new loop.

Reply:
Hello sorry for late reply ... took sort of Friday afternoon off.. I think you are right here I will check again, do some test and get back to you. Thanks